src/services/url.ts

```diff
@@ -22,9 +22,25 @@ class UrlService extends BaseService<UrlService.Options> {
   }
 
   init() {
+    this.beautifier = new UrlBeautifier(this.generateRoutes(), this.opts.beautifier, this.app.config);
     this.readInitialUrl();
   }
 
+  generateRoutes() {
+    const baseTag = WINDOW().document.head.querySelector('base');
+    const routes = this.opts.routes;
+    let baseUrl = baseTag && baseTag.hasAttribute('href') ? baseTag.getAttribute('href') : '/';
```

Comment:
extract this logic (and the next two lines and the `baseTag` declaration) into another method `getBaseUrl()`, this can be a static method on the Service